Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy+PathComponent.swift

```diff
@@ -38,10 +38,13 @@ extension PathHierarchy {
         let full: String
         /// The parsed entity name
         let name: Substring
-        /// The parsed entity kind, if any.
-        var kind: Substring?
-        /// The parsed entity hash, if any.
-        var hash: Substring?
+        /// The parsed disambiguation information, if any.
+        var disambiguation: Disambiguation?
+        
+        enum Disambiguation {
+            /// This path component uses a combination of kind and hash disambiguation
+            case kindAndHash(kind: Substring?, hash: Substring?)
```

Comment:
Would it make sense to add `kind` and `hash` accessor functions or calculated properties in this enum that return `Substring?` That would allow you to further simplify the code that uses these values.

Reply:
My goal is to break apart #643 into 3 pieces: this PR, a follow up PR that changes how disambiguated elements are stored in the path hierarchy, and finally the new link disambiguation syntax. The idea is that we can land the two refactor PRs without changing any behavior and that makes the 3rd (new feature) PR smaller and less likely to have complicated merge conflicts.